cpp/velox/CMakeLists.txt

```diff
@@ -189,8 +189,18 @@ set_target_properties(velox PROPERTIES
     LIBRARY_OUTPUT_DIRECTORY ${root_directory}/releases
     )
 
+## If the folly is not builded in system directories, please add
+## `-DCMAKE_PREFIX_PATH="${folly builded directory}" to the parameters of cmake.
+## It is also applicable to other dependencies.
 find_package(Folly REQUIRED CONFIG)
 find_package(gflags REQUIRED COMPONENTS shared CONFIG)
+find_package(GTest REQUIRED)
+find_package(Protobuf REQUIRED)
```

Comment:
Do we have to find protobuf again? Wasn't it found in cpp/gluten/CMakeLists.txt?

Reply:
Hm, I didn't find it in cpp/gluten/CMakeLists.txt. After my tests, we still need to find it here and include `Protobuf_INCLUDE_DIRS`.

Reply:
Sorry it's in `cpp/core`

https://github.com/oap-project/gluten/blob/a04106623966f8cb384a2c64d89c02fbe18f0eb1/cpp/core/CMakeLists.txt#L116-L124

We had a flag `BUILD_PROTOBUF`. I am not sure whether `find_package(protobuf)` still works when `BUILD_PROTOBUF=ON`?

Reply:
> After my tests, we still need to find it here and include Protobuf_INCLUDE_DIRS.

And I assume if target `gluten` requires protobuf headers publicly in CMake, then the headers can be propagated to target `velox`. You can have a try on it.

Reply:
> We had a flag BUILD_PROTOBUF. I am not sure whether find_package(protobuf) still works when BUILD_PROTOBUF=ON?

Maybe we can use `CACHE PATH` to solve this problem. User can `BUILD_PROTOBUF` to choose whether they want build protobuf, and the variable for protobuf is only configured in core/CMakeLists.txt only once.